department-of-veterans-affairs/caseflow
# Add Hearing Schedule tab to 'test/users' page

Connects #{5710}

### Description
Added Hearing Schedule tab to 'test/users' page.
Added routes to connect url to appropriate controller
Created schedule controller with stubbed method

Also included in this PR is a fix to the Hearings model to properly prefix a class method so it may be invoked from a class method and an instance method. 

### Acceptance Criteria 

### Testing Plan
Tested all tests through the rails console.

To test 'test/users' page

1. Before starting the UI open a rails console and perform the following commands:
`Functions.deny!('System Admin', users: ['BVAJJAST'])`
`Functions.grant!('Hearing Schedule', users: ['BVAJJAST'])`
`Functions.deny!('System Admin', users: ['BVAAABSHIRE'])`
`Functions.grant!('Hearing Prep', users: ['BVAAABSHIRE'])`
2. Start the UI
3. Using BVAJJAST user you should be able to  view the Hearing Schedule main page, but this user will not have access to any of the other product pages.
4. Using BVAAABSHIRE user you should be able to access Hearing Prep page but not Hearing Schedule.

For now the Hearing Schedule main page displays the Out of Service banner until the endpoints are implemented.

To test the Hearings update fix where the mduser is now recorded when doing a db update.

1. In a rails console execute the following command to get a VACOLS hearing row:
`hearing = VACOLS::CaseHearing.find_by(hearing_pkseq: 404074)`
2. Review what the notes1 attribute has.
3. Update the notes1 value with any other value, e.g.
`hearing.update(notes1: 'Testing')`
4. Verify the notes1 column now has the value sent in the update and that no errors were reported.
